kolibri_explore_plugin/collectionviews.py

```diff
@@ -763,9 +766,10 @@ def ensure_initiated(api_function):
     """Decorator to initiate only once in the first API call."""
 
     def wrapper(*args, **kwargs):
-        if _collection_download_manager is None:
-            _read_content_manifests()
-        return api_function(*args, **kwargs)
+        with ENSURE_INITIATED_RLOCK:
+            if _collection_download_manager is None:
+                _read_content_manifests()
+            return api_function(*args, **kwargs)
```

Comment:
Is it intentional that the wrapped function is called with the lock held?

IIUC, the lock is meant to ensure that `_read_content_manifests()` is called only once; so I would expect that the lock could be released before calling the wrapped `api_function`. But I'm probably wrong!

Reply:
Ooh yes, that isn't intentional. Thanks for noticing :)